accepted/00000-auto-generate-bootstrap-repositories.md

```diff
@@ -0,0 +1,71 @@
+- Feature Name: Autogenerate Bootstrap Repositories
+- Start Date: 2020-02-19
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Automatically generate bootstrap repositories
+
+# Motivation
+[motivation]: #motivation
+
+To be able to register a system to Uyuni Server you need to have the software installed which provide the required tooling.
+
+A bootstrap repository provide the software for the supported operating systems with its dependencies. The content for the
+bootstrap repositories come from the normal channels which must be synchronized and updated on regular base.
+
+Users very often do not know this, or forget to create them when they synchronize a new product. Also the bootstrap repository
+is only updated when the user execute the generator script again.
+
+This RFC should describe how to change the current tools to automatically 
+execute the generation of bootstrap repositories after
+a new product was added and after a channel changed when a maintenance update was synced to Uyuni Server.
+
+# Detailed design
+[design]: #detailed-design
+
+The currently exiting tool `mgr-create-bootstrap-repo` list all available bootstrap repositories it could generate.
+This list depends on the products the user has synchronized.
+
+Step 1: provide an option to enable an automatic mode which iterate over all available bootstrap repositories and generate them
+when needed. As we have sometimes have multiple options for the same local "path" we should iterate over the paths to minimize
+the number of runs.
+
+Step 2: add a log file to `mgr-create-bootstrap-repo`. Currently it has no log file. But when this tool is running in the background
+we need the possibility to debug problems. All messages should go to a log file.
+
+Step 3: let `mgr-create-bootstrap-repo` use a configuration file. All commandline options should be possible to set via
+configuration file. This allows to influence the behavior even in automatic mode when it is not possible to change the commandline
+options.
+
+To trigger the re-generation we implement the following algorithm:
```

Comment:
Instead of comparing the last_modified and modified_date, would it make sense for reposync to add a flag (in the DB or even something as simple as touching a file on disk) when the reposync brings new/updated packages? If there are no new/updated packages, then no need to compare anything.

Or might that fail in the case of custom repositories?

Reply:
AFAIK last_modified is only updated when really something changed. So I think this is the flag we look for.
I do not like to add 20 different ways to achieve the same thing.

Of cause I will check if last_modified really only change when a new package was added.

Reply:
Ah OK, perfect then